src/command_modules/azure-cli-batch/azure/cli/command_modules/batch/custom.py

```diff
@@ -41,13 +41,29 @@ def _decorator(func):
 
 
 # Mgmt custom commands
-
 def list_accounts(client, resource_group_name=None):
     acct_list = client.list_by_resource_group(resource_group_name=resource_group_name) \
         if resource_group_name else client.list()
     return list(acct_list)
 
 
+def get_account(cmd, client, resource_group_name=None, account_name=None):
+    if resource_group_name and account_name:
+        return client.get(resource_group_name, account_name)
+    else:
+        account_list = list_accounts(client)
```

Comment:
Move this to behind line 59 so that in the case of `account_endpoint` is missing, and an exception is thrown, there is no redundant operations acted to fetch accounts.